### ISSUE TITLE
$0 for commands should include the command (and parent commands)

### ISSUE THREAD

**Issue by panrafal** [yargs/yargs#998]
If a command has an example, $0 will be replaced with just the binary name. 
This means, that you have to include the command name and it's parent commands in the examples.

This breaks the composability of commands, as they should not know in which context they will be used.

If I have this module:
```js
module.exports = {
  command: 'test',
  builder: (y) => y.example('$0 test', 'just test me')
}
```

And I use it in this context:
```js
command('my-custom-test', '', testModule)
```
The example will show `binary test`, instead of `binary my-custom-test` as it should.

In 10.0 commands started displaying parent commands, now usages and examples should do so too.

**Comment by bcoe** [comment]
@panrafal good catch, I think we should be able to just pull over the logic from the `Commands:` stanza.
